python/lsst/afw/image/exposure.cc

```diff
@@ -20,86 +20,138 @@
  * see <https://www.lsstcorp.org/LegalNotices/>.
  */
 
+#include <memory>
+
 #include <pybind11/pybind11.h>
-//#include <pybind11/operators.h>
 //#include <pybind11/stl.h>
 
+#include "lsst/afw/cameraGeom/Detector.h"
+#include "lsst/afw/image/Calib.h"
+#include "lsst/afw/image/Filter.h"
 #include "lsst/afw/image/Wcs.h"
 #include "lsst/afw/image/Exposure.h"
 #include "lsst/afw/detection/Psf.h"
 
 namespace py = pybind11;
+using namespace pybind11::literals;
+
+namespace lsst {
+namespace afw {
+namespace image {
+
+/**
+Declare a constructor that takes an Exposure of FromPixelT and returns an Exposure cast to ToPixelT
+
+The mask and variance must be of the standard types.
+
+@param[in] cls  The pybind11 wrapper class for the destination exposure
+                (Exposure<ToPixelT, MaskPixel, VariancePixel>)
+*/
+template <typename FromPixelT, typename ToPixelT>
+void declareCastConstructor(py::class_<Exposure<ToPixelT, MaskPixel, VariancePixel>,
+                                       std::shared_ptr<Exposure<ToPixelT, MaskPixel, VariancePixel>>> & cls) {
+    cls.def(py::init<Exposure<FromPixelT, MaskPixel, VariancePixel> const &, bool const>(),
+            "src"_a, "deep"_a);
+}
+
+
+template <typename PixelT>  // only the image type varies; mask and variance are fixed
```

Comment:
I guess I don't understand how the template is instantiated. When you call `declareExposure<int>(mod, "I");`, `PixelT` is obviously `int`, but `Exposure<int, MaskPixel, VariancePixel>` seems like it ought to give an undeclared identifier error.